src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const key in actions[i].extraData) {
+        state[key] = actions[i].extraData[key];
+      }
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (const key of actions[i].keysToRemove) {
+        if (state[key]) {
```

Comment:
you should check is key exist
```suggestion
        if (key in state) {
```